Core/Utilities/TimeLeft/MJFTimeLeft.py

```diff
@@ -39,53 +39,70 @@ def getResourceUsage( self ):
 
     cpuLimit = None
     wallClockLimit = None
+    wallClock = None
+    jobStartSecs = None
 
+    jobFeaturesPath = None
+    machineFeaturesPath = None
+
+    #Getting info from JOBFEATURES
     try:
       # We are not called from TimeLeft.py if these are not set
       jobFeaturesPath = os.environ['JOBFEATURES']
-      machineFeaturesPath = os.environ['MACHINEFEATURES']
     except KeyError:
-      self.log.warn( '$JOBFEATURES and/or $MACHINEFEATURES is not set' )
+      self.log.warn( '$JOBFEATURES is not set' )
 
-    try:
-      wallClockLimit = int( urllib.urlopen( jobFeaturesPath + '/wall_limit_secs' ).read() )
-    except ValueError:
-      self.log.warn( "/wall_limit_secs is unreadable" )
-    except IOError:
-      self.log.warn( "Can't open wall_limit_secs" )
+    if jobFeaturesPath:
+      try:
+        wallClockLimit = int( urllib.urlopen( jobFeaturesPath + '/wall_limit_secs' ).read() )
+      except ValueError:
+        self.log.warn( "/wall_limit_secs is unreadable" )
```

Comment:
For this exception and all the one bellow, would it be interesting to actually print the exception, in order to see why the file could not be opened, or why it could not be cast to an int ?